src/spatial/2d.js

```diff
@@ -81,6 +82,14 @@ Crafty.c("2D", {
      * @see ._attr
      */
     _z: 0,
+
+    /**@
+     * #._globalZ
+     * @comp 2D
+     * When two entities overlap, the one with the larger `_globalZ` will be on top of the other.
+     */
+    _globalZ: null,
```

Comment:
Other alternative would be to listen for `Reorder` events and update `_globalZ` in Renderable accordingly (`z` won't be changed that often in most games). Mouse (similarly to Touch) also implicitly depends on Renderable component (as their common called `findClosestEntityByComponent` function checks for `_visible`).
Don't be fooled by its name, it's not really a spatial property, but rather an imaginary property describing render order (and could be renamed as such).

However, now that I've outlined relevant points, I will be happy either way you decide :)